### ISSUE TITLE
phonegap build Error - Plugin error IOS

### ISSUE THREAD

**Issue by shokry055** [OneSignal/OneSignal-Cordova-SDK#468]
**Description:**

phone gap build error since yesterday

Error - Plugin error (you probably need to remove plugin files from your app): Fetching plugin "onesignal-cordova-plugin" via npm Installing "onesignal-cordova-plugin" at "2.4.6" for ios Failed to install 'onesignal-cordova-plugin': Error: pod: Command failed with exit code 1 at ChildProcess.whenDone (/private/tmp/gimlet/3271732/project/cordova/node_modules/cordova-common/src/superspawn.js:169:23) at ChildProcess.emit (events.js:182:13) at maybeClose (internal/child_process.js:962:16) at Socket.stream.socket.on (internal/child_process.js:381:11) at Socket.emit (events.js:182:13) at Pipe._handle.close (net.js:611:12) pod: Command failed with exit code 1 - You can fix this

**Environment**

<gap:plugin name="onesignal-cordova-plugin" source="npm" />

**Steps to Reproduce Issue:**

  1. error building the application

**Comment by shokry055** [comment]
`"plugman-151 install --platform ios --project /project --plugin onesignal-cordova-plugin --nohooks=.": Fetching plugin "onesignal-cordova-plugin" via npm
Installing "onesignal-cordova-plugin" at "2.4.6" for ios
Failed to install 'onesignal-cordova-plugin': Error: pod: Command failed with exit code 1
    at ChildProcess.whenDone (/private/project/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Socket.stream.socket.on (internal/child_process.js:381:11)
    at Socket.emit (events.js:182:13)
    at Pipe._handle.close (net.js:611:12)
pod: Command failed with exit code 1`

**Comment by FitterX** [comment]
Hi, i am have this erro too...
Anyone have solution?

**Comment by FitterX** [comment]
> `"plugman-151 install --platform ios --project /project --plugin onesignal-cordova-plugin --nohooks=.": Fetching plugin "onesignal-cordova-plugin" via npm Installing "onesignal-cordova-plugin" at "2.4.6" for ios Failed to install 'onesignal-cordova-plugin': Error: pod: Command failed with exit code 1 at ChildProcess.whenDone (/private/project/cordova/node_modules/cordova-common/src/superspawn.js:169:23) at ChildProcess.emit (events.js:182:13) at maybeClose (internal/child_process.js:962:16) at Socket.stream.socket.on (internal/child_process.js:381:11) at Socket.emit (events.js:182:13) at Pipe._handle.close (net.js:611:12) pod: Command failed with exit code 1`

You have found solution?

**Comment by UbySoft** [comment]
Hi, i am have this error: 
phone gap build error

`Error - Plugin error (you probably need to remove plugin files from your app): Fetching plugin "onesignal-cordova-plugin@^2.4.1" via npm Installing "onesignal-cordova-plugin" at "2.4.6" for ios Failed to install 'onesignal-cordova-plugin': Error: pod: Command failed with exit code 1 at ChildProcess.whenDone (/private/tmp/gimlet/3411296/project/cordova/node_modules/cordova-common/src/superspawn.js:169:23) at ChildProcess.emit (events.js:182:13) at maybeClose (internal/child_process.js:962:16) at Socket.stream.socket.on (internal/child_process.js:381:11) at Socket.emit (events.js:182:13) at Pipe._handle.close (net.js:611:12) pod: Command failed with exit code 1`

Anyone have solution?

**Comment by ayslanr** [comment]
Onesignal add cocoapod support in 2.4.6 version launched recently.

I just set the **2.4.5** version and worked for me.

<gap:plugin name="onesignal-cordova-plugin" **spec="2.4.5"** source="npm" />

**Comment by UbySoft** [comment]
> Onesignal add cocoapod support in 2.4.6 version launched recently.
> 
> I just set the **2.4.5** version and worked for me.
> 
> <gap:plugin name="onesignal-cordova-plugin" **spec="2.4.5"** source="npm" />

Thanks ayslanr! ... it works fine for me !

**Comment by vousys** [comment]
> Onesignal add cocoapod support in 2.4.6 version launched recently.
> 
> I just set the **2.4.5** version and worked for me.
> 
> <gap:plugin name="onesignal-cordova-plugin" **spec="2.4.5"** source="npm" />

It works fine for us, but would be fantastic if  they could fix it to use always the latest release.

**Comment by rgomezp** [comment]
Hello,
Thank you for bringing this to our attention. We will look at what may be happening and fix it in the next release.

**Comment by ayslanr** [comment]
> Hello,
> Thank you for bringing this to our attention. We will look at what may be happening and fix it in the next release. Please use the previous version in the meantime.

Yes please!

In my latest update I've forced to downgrade to v2.4.0 to have success in online PhoneGap Build.

**Comment by slack0129** [comment]
The version 2.4.6 is the only version that is supporting iOS 12 so far according to OneSignal's site, they suggest using the latest version which on npm is 2.4.6. I was able to use v2.4.5 just fine to compile and generate the .ipa file via PGB, but we need to be able to use 2.4.6 in order for everything to be working properly in iOS 12.

**Comment by ffrhtt** [comment]
You need to remove cordova plugin (if use phonegap):

`cordova plugin rm onesignal-cordova-plugin`

**Comment by jkasten2** [comment]
There seems to be an issue with Adobe's `plugman` on adding the Cocoapods part of this plugin when using 2.4.6. This issue isn't present when using Cordova, Ionic, or local builds with PhoneGap. It is specific to building on Adobe's PhoneGap Build (PGB) cloud server. The issue seems to only be present in the Cordova 2.4.6 version, 2.4.5 and older build just fine.

We were able to reproduce the issue with PGB builds but haven't been able to find the root cause. It is possible something was something isn't present that PGB needs to add the Cocoapods package, so the next release of this plugin may fix the issue. I'll report here once we release the next version.

**Comment by slack0129** [comment]
@jkasten2 What exactly are we doing wrong then? I have built with version 2.4.5 of this plugin and ran side by side with an Android device, configured the same exact way. The Android device received every push notification perfectly throughout the day, the iOS 12 device did not get a single push notification. At the very moment of this writing I have our app installed on my iPhone running iOS 12 with onesignal version 2.4.5 and still not getting a single notification. We've had the Enable iOS 12 Direct to History enabled the whole time also. Any help would be great!

**Comment by rgomezp** [comment]
@slack0129 ,
How are you building? Are you using PGB or local building?

**Comment by slack0129** [comment]
@rgomezp This originally was a DevExtreme project, once DevExpress dismantled their Native Package Build via Visual Studio (local building) we switched over to the only option which was Adobe PGB on their site at: https://build.phonegap.com. I know we could rewrite this app without DevExtreme at all, but I was tasked with making basic modifications and releasing without doing a complete rewrite. Everything else seems to have turned out just fine, but this onesignal issue has plagued our app and iOS 12 users for a long time now. It's weird that v2.4.5 is supposed to support iOS 12 and I have yet to see it actually work as expected after being built properly with v2.4.5

**Comment by rgomezp** [comment]
@slack0129 
I'm sorry. I understand the frustration. As jkasten2 mentioned, it is not clear what the issue is at this point (and whether it is a problem with their build system or with OneSignal). We will try to resolve this as soon as possible.

**Comment by davequested** [comment]
We are also seeing this issue. If we use this in PGB, it compiles fine:
<gap:plugin name="onesignal-cordova-plugin" source="npm" version="2.4.5" />

This doesn't:
<gap:plugin name="onesignal-cordova-plugin" source="npm" version="2.4.7" />

**Comment by jkasten2** [comment]
I created a ticket about the `plugman-151` error on Adobe's forums however I haven't been able to get someone from Adobe to take a look. Please upvote this thread and reply if there is any additional information you can provide to the ticket.
https://forums.adobe.com/message/11052464#11052464

Thanks.

**Comment by marensas** [comment]
This is a really important issue. This plugin will be useless with iOS 13 release on **September 19th** !!

Please see: https://onesignal.com/blog/ios-13-introduces-4-breaking-changes-to-notifications/

**Comment by youradds** [comment]
> 
> 
> This is a really important issue. This plugin will be useless with iOS 13 release on **September 19th** !!
> 
> Please see: https://onesignal.com/blog/ios-13-introduces-4-breaking-changes-to-notifications/

Seems like no one is interested :/

**Comment by rgomezp** [comment]
Hello everyone,
I understand your frustrations. However, this is an issue with Adobe. If you want to get this resolved, you need to upvote the [issue](https://forums.adobe.com/message/11052464#11052464). Given there is only one other upvote than mine, there is likely not enough pressure for them to prioritize this. Alternatively, you can do a local build, or migrate to regular Cordova.

**Comment by vousys** [comment]
Can you explain in detail, how is that other options to use it, please @rgomezp ?

**Comment by rgomezp** [comment]
Just to clarify, the iOS 13 changes outlined [here](https://onesignal.com/blog/ios-13-introduces-4-breaking-changes-to-notifications/) won't have any negative impact on anyone if PGB hasn't switched to Xcode 11 (our SDK will still work with iOS 13)

**Comment by youradds** [comment]
@rgomezp  thanks. By upvote, do you mean like the post? Or am I missing something?

**Comment by rgomezp** [comment]
@youradds , yes that's what I mean

**Comment by ganeshverma** [comment]
> Onesignal add cocoapod support in 2.4.6 version launched recently.
> 
> I just set the **2.4.5** version and worked for me.
> 
> <gap:plugin name="onesignal-cordova-plugin" **spec="2.4.5"** source="npm" />

Thanks, its working.

**Comment by youradds** [comment]
> 
> 
> > Onesignal add cocoapod support in 2.4.6 version launched recently.
> > I just set the **2.4.5** version and worked for me.
> > <gap:plugin name="onesignal-cordova-plugin" **spec="2.4.5"** source="npm" />
> 
> Thanks, its working.

I'm not sure that is a solution though. It will be using the old push notification methods, and thus won't work with the new Apple updates in their methodology?

**Comment by rgomezp** [comment]
@youradds ,
please see this comment: https://github.com/OneSignal/OneSignal-Cordova-SDK/issues/468#issuecomment-536708847

**Comment by spinninghamster** [comment]
If you have a Mac available, I strongly suggest to migrate to the CLI version. It can be a bit more confusing at first but you'll have much more control over things.

**Comment by youradds** [comment]
Yeah I do have a Mac, and did set it up so I could compile and emulator version (for screenshots). I've never worked out the certificate stuff for the actual builds though (i.e how to link a given certificate to the iOS build), so thats stopped me moving to it fully

**Comment by spinninghamster** [comment]
> Yeah I do have a Mac, and did set it up so I could compile and emulator version (for screenshots). I've never worked out the certificate stuff for the actual builds though (i.e how to link a given certificate to the iOS build), so thats stopped me moving to it fully

That should be quite easy. Assuming you created the project with Cordova (there are guides), and added the platforms iOS and Android (again, good guides available), just open the XCode project, login with your Apple Developer Account and select to manually manage certificates. XCode will fetch the certificates for you from your account.

**Comment by shokry055** [comment]
use This version For IOS under 
   <platform name="ios">
   <gap:plugin name="onesignal-cordova-plugin" spec="2.4.5" source="npm" />

under 
    <platform name="android">
    USE
    <gap:plugin name="onesignal-cordova-plugin" source="npm" />
it will install last V on android And IOS will Build successfully

**Comment by realyze** [comment]
I have the same issue building a Meteor app
```
While adding plugin onesignal-cordova-plugin@2.8.0 to Cordova project:          
   Error: pod: Command failed with exit code 31 Error output:                      
   Ignoring bigdecimal-1.4.3 because its extensions are not built.  Try: gem pristine bigdecimal --version 1.4.3
   Ignoring io-console-0.4.7 because its extensions are not built.  Try: gem pristine io-console --version 0.4.7
   Ignoring libxml-ruby-3.1.0 because its extensions are not built.  Try: gem pristine libxml-ruby --version 3.1.0
   Ignoring psych-3.1.0 because its extensions are not built.  Try: gem pristine psych --version 3.1.0
   at ChildProcess.whenDone
```

I'm not quite sure how that would be related to Adobe PhoneGap build. I'm building the app locally on a macbook pro. 🤔

**Comment by canado** [comment]
I have the same problem when I use ver 2.8.1 of onesignal ( the latest to date )

**Comment by canado** [comment]
any updates @onesignal team?

**Comment by jkasten2** [comment]
@canado 
Chris_W__Griffith at Adobe said the following about the PGB build issue:
> I would strongly suggest that you begin migrating your builds to your own systems. Adobe has placed PhoneGap Build in maintenance status, so do not expect any fast resolutions to issues like this.

https://community.adobe.com/t5/phonegap-build/plugman-151-error-with-pod/m-p/10645685#M8379

I just pinged them again on their forum if they can help debug the issue so we are waiting on them to help solve this PGB specific compatibility issue.

**Comment by canado** [comment]
@jkasten2  Chris_W is always saying the same thing, but with projects dating from 2013 it is not helping a lot. 
Still no further info feedback?

**Comment by mredbishop** [comment]
It worked with 2.4.5 then it worked again with 2.7.0 so there must be some correlation as to what broke it in between. Seems like it can't be just Adobe if there are two disparate versions working? I could be wrong of course and probably am :o)

**Comment by jkasten2** [comment]
> @jkasten2 Chris_W is always saying the same thing, but with projects dating from 2013 it is not helping a lot.
> Still no further info feedback?

Ah, mistook ["Adobe Community Professional"](https://community.adobe.com/t5/community-help/meet-the-experts/td-p/4788640) as someone who works for Adobe so he isn't officially stating deperacation of PGB.

**Comment by jkasten2** [comment]
> It worked with 2.4.5 then it worked again with 2.7.0 so there must be some correlation as to what broke it in between. Seems like it can't be just Adobe if there are two disparate versions working? I could be wrong of course and probably am :o)

The issue seems to be specifically related to `plugman`, with PGB they use a specific fork `plugman-151`. `plugman` isn't used by default when building locally with Cordova, PGB CLI, Ionic, etc which doesn't have this issue. I have attempted to use `plugman` locally but ran into an issue getting it setup correctly to even start reproducing the PGB issue. We will take another pass at it to see if we can get `plugman` working locally.

Thanks for the detail on 2.7.0, we will take a closer look at what changed between 2.4.5 -> 2.4.6 and 2.6.0 -> 2.7.0.

Anyone else running into this PGB issue I recommend locking to 2.7.0 as @mredbishop suggests until we can resolve this issue.

**Comment by canado** [comment]
Hello Josh

Do you have any feedback about the problem?
I can't compile and submit my app since end of November 2019!

Regards

**Comment by jkasten2** [comment]
@canado `onesignal-cordova-plugin@2.7.0` supports iOS 13 and this should be the one to use if you need to build on PGB. You noted you were getting "Apns Delegate Never Fired" when using this version,  there is a known issue with iOS where this can happen to some devices in some cases. Toggle wifi fixes this issue, it may only be development related. https://github.com/OneSignal/OneSignal-iOS-SDK/issues/514

The lasted `onesignal-cordova-plugin` can be used if you use PhoneGap CLI locally. The Cocoapods issue is specifically related plugman which PGB uses.

**Comment by canado** [comment]
Hello @jkasten2 thank you for the follow-up, I was using the 2.8.2 because the 2.7.0 was not working.
I saw the trick of toggling on and off the wi-fi but unfortunately, this is not working.
Now I am not confident in releasing a live version if it is not working in a dev environment. I am scared that my previous iOS users won't have access to push notifications.

Is there a way you can confirm that it will work?
And I am not confident in building locally since all my setup / config were done using PGB : (

Regards

**Comment by slack0129** [comment]
All, just wanted to circle back here. I saw your comments and was able to successfully build on Adobe PhoneGap Build site for iOS with version 2.7.0 however, @jkasten2  you can read for yourself here: https://www.npmjs.com/package/onesignal-cordova-plugin/v/2.7.0 that 2.7.0 only supports up to iOS 12.2 when in fact we've been dealing with iOS 13 now for sometime. Furthermore, I find it extremely odd that there was a successful version at 2.4.5 that worked, and then again at 2.7.0 but this is all entirely on Adobe as their problem to resolve. As it would seem, OneSignal is able to produce a version that for some reason or another works flawlessly with Adobe PhoneGap Build site, yet fails on countless versions in between and since 2.7.0. I would think, as if this were my product, that the OneSignal team would be able to go about comparing the 2.4.5 and the 2.7.0 versions versus the other non working versions to see if there might be some kind change that could be made to resolve this similar to whatever worked in 2.4.5 and 2.7.0 versions. 

Oddly enough, I was able to get version 2.8.3, the latest build to successfully work for both Android and iOS in my latest go round with the Adobe PhoneGap Build site. This failed numerous times for me with the same Cocoa pods error we've been seeing, not to mention having to reupload code everytime I would switch the OneSignal plugin version, but @canado your comments about toggling the wifi during the build and how that used to work to trip something up and generate an IPA successfully, got me thinking. So I set a build up using the following in my config: <plugin name="onesignal-cordova-plugin" source="npm" /> My company uses Cisco's AnyConnect VPN product to VPN to our network if need be. I started the build on the site and after some seconds looking at the spinning of the builds running, I switched over to the VPN connection, the APK file completed first as expected, the iOS version spun and spun, but it did successfully generate an IPA file, I have verified this working with both our dev and production signing certificates. See screenshot of log from the site.

![image](https://user-images.githubusercontent.com/29146920/74383706-bc357880-4dbd-11ea-87fb-f649ffef18a0.png)

**Comment by spinninghamster** [comment]
Just a suggestion. If you're in any way able to, I recommend everyone to switch to the CLI version. PhoneGap is not being maintained anymore and I have a feeling that eventually it will go away or become unusable, as in this case.

**Comment by canado** [comment]
Good morning @slack0129 thank you very much for your investigation.
The only small detail and people on this thread can correct me if I am wrong, but I thought that the toggle wi-fi trick was on the client-side, once the app was installed and running, and not during compilation.

And yes @slack0129 you are correct I am a bit stuck since November by not having a single version that is working for both iOS 12.X and iOS 13

@spinninghamster I am afraid that you are right, but do you know if it would be easy to change from Phonegap build to Phonegap CLI, and is there any insurances that it will work?

Thanks

**Comment by spinninghamster** [comment]
> @spinninghamster I am afraid that you are right, but do you know if it would be easy to change from Phonegap build to Phonegap CLI, and is there any insurances that it will work?

Yes, it's quite simple. I did it for several projects. If there are errors, it will tell you what they are but there should be no major ones. Maybe some paths that need to be changed but that's about it. I recommend you install everything first that is needed for CLI (read some of the getting started guides) and then launch a test app with CLI. You can then compare the config.xml files and see what the differences are, adjust it, etc. Then simply add all the plugins with command line and it should work fine.

You will find that using CLI gives you much more freedom and less headaches than PhoneGap Build.

**Comment by canado** [comment]
I am trying to build my app for iOS on my windows using PhoneGap CLI is there a way to make this work?

**Comment by jkasten2** [comment]
@canado Unfortunately due to Apple's Xcode and xcodebuild command only being available on macOS you must use a mac to build for iOS.

**Comment by canado** [comment]
Ok so this means I am stuck? : (

**Comment by canado** [comment]
Can someone assure me that if I use a mac it will work, ( compiling onesignal and have everything working just fine? )

Thanks

**Comment by spinninghamster** [comment]
> Can someone assure me that if I use a mac it will work, ( compiling onesignal and have everything working just fine? )
> 
> Thanks

You should not have any problems with a Mac. Maybe you can use a friends' Mac or find someone who has one locally (at a developer meet up)...

**Comment by canado** [comment]
Hi @spinninghamster 

I managed to run xcode on a VM
after putting back my project on again ( updating scripts etc ... )
I can now publish from xcode

I tried the manual setup for the onesignal as well as the automatic https://onesignal.com/provisionator

still in both cases I have 
Apns Delegate Never Fired
And again I have been playing with the wi-fi

**Comment by canado** [comment]
Any updates on this, I still have Apns Delegate Never Fired : (

**Comment by widonwaker** [comment]
update: phonegap build fails most of the times due to 'framework type="podspec" deprecated. The problem is this plugin also fails on Cordova CLI by Terminal. I tried it on a fresh new Cordova project with only this plugin installed and it fails.

**Comment by jkasten2** [comment]
@widonwaker The "Apns Delegate Never Fired" isn't related to this issue, please open a new issue with more details.
@widonwaker You should be able to add this OneSignal plugin via Cordova CLI. Please open a new issue with the exact Cordova version you are using along with the full error you are seeing.

**Comment by jkasten2** [comment]
PhoneGap and PhoneGap Build (PGB) will no longer be supported by Adobe on October 1, 2020
https://blog.phonegap.com/update-for-customers-using-phonegap-and-phonegap-build-cc701c77502c

Since PhoneGap Build (PGB) will be shutting down relatively soon we are closing this issue as "won't fix".
